File: C++/Varshmaan/MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int count=0;
        if(nums.size()==1){
            return nums[0];
        }
        else{
        for(int i=1;i<nums.size();i++){
            if(nums[i]==nums[i-1]){
                count++;
                if(count>=(nums.size()/2)){
                    return nums[i];
                }
            }
            else{
                count=0;
            }
            
        }
        }
        return 0;
    }
};

File: Python/Arrays/MajorityElement.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n=len(nums)
        cache={} #key=num : value= count
        
        for i in nums: #keeping count of occurence of every element
            if i in cache.keys():
                cache[i]+=1
            else:
                cache[i]=1
        
        a=list(cache.keys()) #making a list of all distinct elements
        for j in a:
            if cache[j]>int(n/2): #we return the element whose count is greater than n/2
                return j
        return -1